#**DATA MATH 361 Lab Assignment 11** by John Dominguez-Trujillo

*ALL CODE WAS DEVELOPED IN VISUAL STUDIO CODE AND PASTED INTO GOOGLE COLAB*

###**Activity 1:** Write Python code for a regression table of 'Adjusted Costs (billions USD 2024)' against 'Year'. Write Python code for a regression table of 'Adjusted Costs (billions USD 2024)' against 'US Pop (millions)'. Write Python code for a regression table of 'Adjusted Costs (billions USD 2024)' against 'Year' and 'US Pop (millions)'. For each of the three regressions, report whether the variable(s) is/are significant predictors of hurrican damage. Is there an apparent conflict between what you observe in the three regressions?

In [1]:
### Activity 1

## IMPORTS ESSENTIAL LIBRARIES
## PANDAS IS A LIBRARY USED FOR DATA MANIPULATION, ANALYSIS, AND DATA STRUCTURES
## STATSMODELS IS A LIBRARY USED FOR ESTIMATING STATISTICAL MODELS AND PERFORMING REGRESSION ANALYSIS, INCLUDING GENERATING REGRESSION TABLES
## STATS MODELS FORMULA API (SMF) ALLOWS LINEAR MODELS TO BE SPECIFIED USING R-LIKE FORMULAS
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

## LOADS DATA FROM CSV INTO A VARIABLE
## USING A URL THAT LINKS DIRECTLY TO THE CSV FILE IN MY GITHUB SO ANYONE CAN ACCESS IT
url = 'https://raw.githubusercontent.com/JohnDT-MechE/DATA-MATH-361-Lab-Assignments/refs/heads/main/Lab_Assignment_11/Hurricanes%202024%20data.csv'
df = pd.read_csv(url)

## THIS SELECTS SPECIFIC COLUMNS FROM THE DATAFRAME AND REMOVES ANY ROWS THAT CONTAIN MISSING VALUES
df = df[["Year", "US Pop (millions)", "Adjusted Costs (billions USD 2024)"]].dropna()


## 1.1 REGRESSION TABLE OF ADJUSTED COSTS AGAINST YEAR

## THIS USES STATSMODELS FORMULA API TO FIT A LINEAR REGRESSION MODEL PREDICTING ADJUSTED COSTS BASED ON YEAR
## .FIT() TRAINS THE MODEL ON THE DATA
model1 = smf.ols('Q("Adjusted Costs (billions USD 2024)") ~ Year', data=df).fit()

## PRINTS A TITLE LINE TO IMPROVE READABILITY OF OUTPUT
print("\n--- Regression 1: Adjusted Costs vs. Year ---")

## PRINTS THE FULL REGRESSION TABLE TO THE OUTPUT TERMINAL
print(model1.summary())


## 1.2 REGRESSION TABLE OF ADJUSTED COSTS AGAINST US POPULATION

## THIS USES STATSMODELS FORMULA API TO FIT A LINEAR REGRESSION MODEL PREDICTING ADJUSTED COSTS BASED ON US POPULATION
## DIFFERENCE IS THAT THE PREDICTOR IS NOW US POPULATION
model2 = smf.ols('Q("Adjusted Costs (billions USD 2024)") ~ Q("US Pop (millions)")', data=df).fit()
print("\n--- Regression 2: Adjusted Costs vs. US Population ---")
print(model2.summary())


## 1.2 REGRESSION TABLE OF ADJUSTED COSTS AGAINST YEAR AND US POPULATION

## THIS USES STATSMODELS FORMULA API TO FIT A LINEAR REGRESSION MODEL PREDICTING ADJUSTED COSTS BASED ON YEAR AND US POPULATION
## DIFFERENCE IS THAT THE PREDICTOR IS NOW YEAR AND US POPULATION
model3 = smf.ols('Q("Adjusted Costs (billions USD 2024)") ~ Year + Q("US Pop (millions)")', data=df).fit()
print("\n--- Regression 3: Adjusted Costs vs. Year and US Population ---")
print(model3.summary())


--- Regression 1: Adjusted Costs vs. Year ---
                                       OLS Regression Results                                      
Dep. Variable:     Q("Adjusted Costs (billions USD 2024)")   R-squared:                       0.080
Model:                                                 OLS   Adj. R-squared:                  0.063
Method:                                      Least Squares   F-statistic:                     4.610
Date:                                     Sun, 20 Apr 2025   Prob (F-statistic):             0.0364
Time:                                             18:34:00   Log-Likelihood:                -278.54
No. Observations:                                       55   AIC:                             561.1
Df Residuals:                                           53   BIC:                             565.1
Df Model:                                                1                                         
Covariance Type:                                 nonr

**Regression Table of Adjusted Costs Against Year:** The p-value for year is equal to 0.036. Since 0.036 is less than 0.05, year is a statistically significant predictor of hurricane damage costs. In other words, as time progresses, hurricane damage costs tend to increase. This can possibly be due to inflation, urban expansion, and/or climate change impacts.

**Regression Table of Adjusted Costs Against U.S. Population:** The p-value for U.S. Population is 0.031. Since 0.031 is less than 0.05, U.S. Population is a statistically significant predictor of hurricane damage costs. In other words, as the U.S. Population increases, hurricane damage costs tend to increase. Higher population levels may correlate with greater infrastructure and property exposure to hurricane damage, which could possible be leading to higher costs.

**Regression Table of Adjusted Costs Against Year and U.S. Population:** The p-value for year is equal to 0.300 and the p-value for U.S. Population is equal to 0.242. Since both variables are greater than 0.05, they are not statistically significant when included together. Since neither predictor is significant when both are in the model, this could suggest multicollinearity meaning that the variables are highly correlated (since both increase over time), making it hard to distinguish their individual effects.

**Apparent Conflict:** Yes, there seems to be an apparent conflict. In the first two regression models, year and U.S. Population each appeared to be significant predictors on their own, however, when combined in the third regression model, neither remained significant. As stated briefly before, this conflict likely arises due to multicollinearity since both year and U.S. Population are strongly related (since they both increase over time). Due to this, their individual contributions to predicting hurricane damage costs are not distinguishable when included in the same model. This is further supported by the very large condition number of 2.85e+06 in the third regression model, which signals multicollinearity issues. In conclusion, both year and U.S. Population are significant individual predictors of adjusted hurricane damage costs. However, when included together in the same regression model, neither one remains significant, which is an indicator of multicollinearity issues. This suggests that while each variable is associated with damage cost increases, their strong correlation prevents clear seperation of their effects in a combined model.